arctic/store/version_store.py

```diff
@@ -524,9 +568,123 @@ def _publish_change(self, symbol, version):
             mongo_retry(self._changes.insert_one)(version)
 
     @mongo_retry
-    def write(self, symbol, data, metadata=None, prune_previous_version=True, **kwargs):
+    def _write_metadata_history(self, symbol, metadata, start_time, end_time=None, **kwargs):
         """
-        Write 'data' under the specified 'symbol' name to this library.
+        Create a new metadata entry in ._metadata collection
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        metadata : `dict`
+            to be persisted
+        start_time : `datetime.datetime`
+            when entry becomes effective
+        end_time : `datetime.datetime` or None
+            when entry expires. When set to None entry is still in effect
+            Default: None
+        kwargs :
+            passed through to the write handler
+
+        Returns
+        -------
+        VersionedItem containing .metadata
+        """
+        self._arctic_lib.check_quota()
+        version = {'_id': bson.ObjectId()}
+        version['symbol'] = symbol
+        version['metadata'] = metadata
+        version['_start_time'] = start_time
+        if end_time is not None:
+            version['_end_time'] = end_time
```

Comment:
Does anything check whether there's a previous metadata document that has no `end_time`?